### DISCUSSION TITLE
Termux root access

### DISCUSSION THREAD

**Discussion by anonymous008** [termux/termux-app#2917] [Q&A] [answered]
I installed termux because it didn't need device to be rooted. Now I can see that it is asking for root access. When I use `su` or `tsu`, it says, 
`No superuser binary detected. `
`Are you rooted?`
How to fix the issue without rooting my device?

**Comment by agnostic-apollo** [comment, marked as answer]
Termux does not "need" root access to run, unless some package, like from `root-repo` requires it, in which case you need to root your device (or use some vulnerability to escalate privileges). Termux can't magically give you root access.
